File: Clojure/output.clj
;  Output functions for the Clojure versions of the π using Quadrature codes.
;
; Copyright © 2009, 2013  Russel Winder

(defn out [name pi n elapseTime]
  (println (str "==================== " name))
  (println (str "\tπ = " pi))
  (println (str "\titeration count = " n))
  (println (str "\telapse time = " elapseTime)))

(defn outn [name pi n elapseTime numberOfThreads]
  (out name pi n elapseTime)
  (println (str "\tnumber of threads = " numberOfThreads))
  (println (str "\tnumber of processors = " (.. Runtime getRuntime availableProcessors))))

File: Kotlin/src/output.kt
package uk.org.winder.pi_quadrature

fun out(banner:String, pi:Double, n:Int, elapseTime:Double) {
  println("================ " + banner)
  println("\tπ = " + pi)
  println("\titeration count = " + n)
  println("\telapse time = " + elapseTime)
}

fun out(banner:String, pi:Double, n:Int, elapseTime:Double, numberOfTasks:Int) {
  out(banner, pi, n, elapseTime)
  println("\tprocessor count = " + Runtime.getRuntime().availableProcessors())
  println("\tnumber of tasks = " + numberOfTasks)
}
